modular_hearthstone/code/modules/spells/roguetown/wizard.dm

```diff
@@ -1496,6 +1497,50 @@
 	to_chat(target, span_warning("My legs feel remarkably weaker."))
 	target.Immobilize(5)
 
+/obj/effect/proc_holder/spell/invoked/blink
+	name = "Blink"
+	desc = "Teleport to a targeted location within your field of view."
+	school = "conjuration"
+	cost = 1
+	releasedrain = 30
+	chargedrain = 1
+	chargetime = 15
+	charge_max = 20 SECONDS
+	warnie = "spellwarning"
+	no_early_release = TRUE
+	movement_interrupt = FALSE
+	charging_slowdown = 2
+	chargedloop = /datum/looping_sound/invokegen
+	associated_skill = /datum/skill/magic/arcane
+	overlay_state = "blink"
+	xp_gain = TRUE
+
+/obj/effect/proc_holder/spell/invoked/blink/cast(list/targets, mob/user = usr)
+	var/turf/T = get_turf(targets[1])
+	var/turf/start = get_turf(user)
+	
+	if(!T)
+		to_chat(user, span_warning("Invalid target location!"))
+		revert_cast()
+		return
+		
+	// Check if there's a wall in the way, but exclude the target turf
+	var/list/turf_list = getline(start, T)
```

Comment:
There's a can_see proc that would draw a line and establish line of sight to the turf. Alternatively, you could also check if the target turf has the caster in its viewers
(code/_HELPERS/unsorted.dm L:402)